Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -2893,7 +2894,16 @@ public async Task AddVirtualFolder(string name, CollectionTypeOptions? collectio
             var virtualFolderPath = Path.Combine(rootFolderPath, name);
             while (Directory.Exists(virtualFolderPath))
```

Comment:
Performance isn't exactly a concern here, so we could just use [`Directory.GetDirectories`](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.getdirectories?view=net-6.0#System_IO_Directory_GetDirectories_System_String_System_String_System_IO_SearchOption_), with a wildcard pattern (eg. `name + "*"`) and count them?

Reply:
Just counting the matches doesn't account for libraries having been deleted:

Say I create three libraries: "Movies", "Movies 2", and "Movies 3", then delete "Movies 2". If I try to create another library named "Movies" then by count of "Movies*" I would expect to be creating "Movies 3" (which conflicts with an existing library) but by iterating through the next sequential name I find that I can create "Movies 2" instead.

Reply:
That is true. You could simplify it though. Just keep a separate counter and concat it with name until the loop condition is unsatisfied.

Reply:
Yeah, I definitely over thought that originally.